lib/rules/amd-function-arity.js

```diff
@@ -30,10 +34,35 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
```

Comment:
This still seems a bit overwrought for me, and I don't think the extra `allowedExtraDependencies` variable is even necessary. I'm thinking along the lines of:
```js
// This resolves to either `true`, `false`, an `Array` or `undefined`
var allowExtraDependencies = context.options && context.options[0] && context.options[0].allowExtraDependencies;

// Set the default
if (allowExtraDependencies === undefined) {
    allowExtraDependencies = false;
}
```